From 1236a4a1f3490e6f240ef039f52f1e91707b9ace Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Mon, 23 Apr 2012 12:11:26 +0530
Subject: [PATCH 0942/1092] ARM: OMAP2+: am335x: Remove Support for Low Cost
 EVM

Since Low Cost EVM is dropped from development, remove the support as
well in the code.

While here, also correct the coding style in EEPROM Header information.

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c              | 34 +++++-----------------
 arch/arm/mach-omap2/include/mach/board-am335xevm.h | 11 ++++---
 2 files changed, 12 insertions(+), 33 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index b31d5d4..5ca3c8b 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -274,11 +274,10 @@ static struct omap_board_config_kernel am335x_evm_config[] __initdata = {
 *  Header		4	0xAA, 0x55, 0x33, 0xEE
 *
 *  Board Name		8	Name for board in ASCII.
-*				example "A33515BB" = "AM335X
-				Low Cost EVM board"
+*				Example "A33515BB" = "AM335x 15x15 Base Board"
 *
-*  Version		4	Hardware version code for board in
-*				in ASCII. "1.0A" = rev.01.0A
+*  Version		4	Hardware version code for board	in ASCII.
+*				"1.0A" = rev.01.0A
 *
 *  Serial Number	12	Serial number of the board. This is a 12
 *				character string which is WWYY4P16nnnn, where
@@ -289,8 +288,7 @@ static struct omap_board_config_kernel am335x_evm_config[] __initdata = {
 *  Configuration option	32	Codes(TBD) to show the configuration
 *				setup on this board.
 *
-*  Available		32720	Available space for other non-volatile
-*				data.
+*  Available		32720	Available space for other non-volatile data.
 */
 struct am335x_evm_eeprom_config {
 	u32	header;
@@ -1763,15 +1761,6 @@ static void profibus_init(int evm_id, int profile)
 	return;
 }
 
-/* Low-Cost EVM */
-static struct evm_dev_cfg low_cost_evm_dev_cfg[] = {
-	{rgmii1_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
-	{usb0_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
-	{usb1_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
-	{evm_nand_init, DEV_ON_BASEBOARD, PROFILE_NONE},
-	{NULL, 0, 0},
-};
-
 /* General Purpose EVM */
 static struct evm_dev_cfg gen_purp_evm_dev_cfg[] = {
 	{enable_ecap0,	DEV_ON_DGHTR_BRD, (PROFILE_0 | PROFILE_1 |
@@ -1857,13 +1846,6 @@ static struct evm_dev_cfg beaglebone_dev_cfg[] = {
 	{NULL, 0, 0},
 };
 
-static void setup_low_cost_evm(void)
-{
-	pr_info("The board is a AM335x Low Cost EVM.\n");
-
-	_configure_device(LOW_COST_EVM, low_cost_evm_dev_cfg, PROFILE_NONE);
-}
-
 static void setup_general_purpose_evm(void)
 {
 	u32 prof_sel = am335x_get_profile_selection();
@@ -2036,9 +2018,7 @@ static void am335x_evm_setup(struct memory_accessor *mem_acc, void *context)
 		snprintf(tmp, 7, "%s", config.opt);
 		pr_info("SKU: %s\n", tmp);
 
-		if (!strncmp("SKU#00", config.opt, 6))
-			setup_low_cost_evm();
-		else if (!strncmp("SKU#01", config.opt, 6))
+		if (!strncmp("SKU#01", config.opt, 6))
 			setup_general_purpose_evm();
 		else if (!strncmp("SKU#02", config.opt, 6))
 			setup_ind_auto_motor_ctrl_evm();
@@ -2216,8 +2196,8 @@ static void evm_init_cpld(void)
 
 static void __init am335x_evm_i2c_init(void)
 {
-	/* Initially assume Low Cost EVM Config */
-	am335x_evm_id = LOW_COST_EVM;
+	/* Initially assume General Purpose EVM Config */
+	am335x_evm_id = GEN_PURP_EVM;
 
 	evm_init_cpld();
 
diff --git a/arch/arm/mach-omap2/include/mach/board-am335xevm.h b/arch/arm/mach-omap2/include/mach/board-am335xevm.h
index a8fe93a..a198fe8 100644
--- a/arch/arm/mach-omap2/include/mach/board-am335xevm.h
+++ b/arch/arm/mach-omap2/include/mach/board-am335xevm.h
@@ -20,12 +20,11 @@
 #define DAUG_BOARD_I2C_ADDR	0x51
 #define LCD_BOARD_I2C_ADDR	0x52
 
-#define LOW_COST_EVM		0
-#define GEN_PURP_EVM		1
-#define IND_AUT_MTR_EVM		2
-#define IP_PHN_EVM		3
-#define BEAGLE_BONE_OLD		4
-#define BEAGLE_BONE_A3		5
+#define GEN_PURP_EVM		0
+#define IND_AUT_MTR_EVM		1
+#define IP_PHN_EVM		2
+#define BEAGLE_BONE_OLD		3
+#define BEAGLE_BONE_A3		4
 
 /* REVIST : check posibility of PROFILE_(x) syntax usage */
 #define PROFILE_NONE	-1	/* Few EVM doesn't have profiles */
-- 
1.7.11.2

